cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -264,16 +264,15 @@ else()
                  ${ARROW_BOOST_BUILD_VERSION})
   set_urls(
     BOOST_SOURCE_URL
+    "https://dl.bintray.com/boostorg/release/${ARROW_BOOST_BUILD_VERSION}/source/boost_${ARROW_BOOST_BUILD_VERSION_UNDERSCORES}.tar.gz"
+    "https://github.com/boostorg/boost/archive/boost-${ARROW_BOOST_BUILD_VERSION}.tar.gz"
```

Comment:
```suggestion
```

And if you want to bump the others up, put them after the ursalabs/arrow-boost one.